### PR TITLE
tests: new tests folders organization

### PR DESCRIPTION
**Pull request by Cristhianzl**
## Overview
This pull request introduces a new folder organization for our test suite, conceptually aligned with updated documentation on test structure and purpose.

## Changes
- Restructured test folders to reflect the new organizational concept

## New Folder Structure
- **Core**: Critical system tests
  - Key functionalities
  - Critical bug fixes
  - Main template examples
  - Complex component unit tests

- **Extended**: Less critical system tests
  - Specific functionalities
  - General bugs
  - Simple integrations
  - Secondary features
  - Simple component unit tests

### Subcategories
- Bugs
- Integrations
- Features
- Unit
- Assets

## Benefits
- Improved test organization and clarity
- Better alignment between tests and system priorities
- Easier navigation and maintenance of the test suite
- Clear documentation for current and future developers

### CODE REVIEW

**Review by anovazzi1 — APPROVED** [review]
lgtm

File: src/frontend/tests/core/bugs/generalBugs-shard-4.spec.ts
Comment (jordanrfrazier) [inline src/frontend/tests/core/bugs/generalBugs-shard-4.spec.ts:1]: Would suggest calling this folder `/core/regression`

File: src/frontend/tests/extended/bugs/generalBugs-shard-0.spec.ts
Comment (jordanrfrazier) [inline src/frontend/tests/extended/bugs/generalBugs-shard-0.spec.ts:1]: same

File: src/frontend/tests/extended/unit/chatInputOutput.spec.ts
Comment (jordanrfrazier) [inline src/frontend/tests/extended/unit/chatInputOutput.spec.ts:1]: Curious if we need an `extended/unit` section - I would think that all `unit` tests are short-running and negligible cost to add to `core`

**Review by jordanrfrazier — APPROVED** [review]

**Review by NadirJ — APPROVED** [review]

File: src/frontend/tests/core/bugs/generalBugs-shard-4.spec.ts
Comment (Cristhianzl) [reply src/frontend/tests/core/bugs/generalBugs-shard-4.spec.ts:1]: Seems good @jordanrfrazier I'll update this

File: src/frontend/tests/extended/unit/chatInputOutput.spec.ts
Comment (Cristhianzl) [reply src/frontend/tests/extended/unit/chatInputOutput.spec.ts:1]: yep, I just moved the more complex ones to the core folders.. maybe in the future, if we need to optimize, we can make changes to improve performance..

**Pull request merged**